src/main/scala/cromwell/binding/Workflow.scala

```diff
@@ -140,7 +140,24 @@ case class Workflow(name: String,
    *         of all outputs from all `call`s within this workflow
    */
   def outputs: Map[FullyQualifiedName, WdlType] = {
-    val outputs = for (call <- calls; output <- call.task.outputs) yield (s"${call.fullyQualifiedName}.${output.name}", output.wdlType)
-    outputs.toMap
+    val outputs = for {
+      call: Call <- calls
+      output <- call.task.outputs
+    } yield (s"${call.fullyQualifiedName}.${output.name}", output.wdlType)
+
+    if(workflowOutputDecls isEmpty) {
+      outputs toMap
+    }
+    else {
+      outputs filter {
+        case (fqn, wdlType) =>
+          val x = fqn
+          val y = workflowOutputDecls
+          y.isEmpty || y.exists(_.outputMatchesDeclaration(x))
```

Comment:
do you need the x & y? in particular, x is only used once and is only 2 chars shorter than fqn